src/components/application_manager/test/commands/mobile/set_icon_reques_test.cc

```diff
@@ -0,0 +1,193 @@
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdint.h>
+#include <string>
+
+#include "gtest/gtest.h"
+#include "utils/shared_ptr.h"
+#include "utils/file_system.h"
+#include "commands/command_request_test.h"
+#include "application_manager/mock_application_manager.h"
+#include "application_manager/mock_message_helper.h"
+#include "application_manager/mock_hmi_capabilities.h"
+#include "mobile/set_icon_request.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace set_icon_request {
+
+namespace am = ::application_manager;
+namespace mobile_result = mobile_apis::Result;
+
+using ::testing::Mock;
+using ::testing::_;
+
+using am::commands::SetIconRequest;
+using am::commands::MessageSharedPtr;
+
+typedef ::utils::SharedPtr<SetIconRequest> CommandPtr;
+
+namespace {
+const uint32_t kConnectionKey = 1u;
+const std::string kFileName = "file.txt";
+}  // namespace
+
+MATCHER_P(CheckMsgType, msg_type, "") {
+  return msg_type ==
+         static_cast<int32_t>(
+             (*arg)[am::strings::params][am::strings::message_type].asInt());
+}
+
+class SetIconRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ protected:
+  void SetUp() OVERRIDE {
+    ON_CALL(mock_app_manager_, get_settings())
+        .WillByDefault(ReturnRef(mock_app_manager_settings_));
+  }
+  void TearDown() OVERRIDE {
+    if (file_system::FileExists("./" + kFileName)) {
+      EXPECT_TRUE(file_system::DeleteFile("./" + kFileName));
+    }
+  }
+};
+
+TEST_F(SetIconRequestTest, Run_InvalidApp_UNSUCCESS) {
+  MessageSharedPtr msg(CreateMessage(smart_objects::SmartType_Map));
+  (*msg)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+  CommandPtr command(CreateCommand<SetIconRequest>(msg));
+  MockAppPtr invalid_mock_app;
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(invalid_mock_app));
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(
+          MobileResultCodeIs(mobile_result::APPLICATION_NOT_REGISTERED), _));
+
+  command->Run();
+}
+
+TEST_F(SetIconRequestTest, Run_InvalidFile_UNSUCCESS) {
+  MessageSharedPtr msg(CreateMessage(smart_objects::SmartType_Map));
+  (*msg)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+  (*msg)[am::strings::msg_params][am::strings::sync_file_name] = kFileName;
+
+  CommandPtr command(CreateCommand<SetIconRequest>(msg));
```

Comment:
@okozlovlux, as for me, better to move `command` and its `message` to fixture, and then init some basic message fields in constructor (Like `connection_key` and `sync_file_name`)